00 SFC Boilerplate/README.md

```diff
@@ -0,0 +1,319 @@
+
+Summary steps:
+
+- Prerequisites: Install Node.js
+- Initialize **package.json** (with `npm init`)
+- Install:
+  - Webpack and webpack-dev-server.
+  - TypeScript.
+  - Bootstrap.
+- Setup **webpack.config.js**
+- Create a test js file.
+- Create a simple HTML file.
+
+# Prerequisites
+
+Install [Node.js and npm](https://nodejs.org/en/) (v6.x.x or higher) if they are not already installed on your computer.
+
+> Verify that you are running at least node v6.x.x and npm 3.x.x by running `node -v` and `npm -v` in a terminal/console window. Older versions may produce errors.
+
+## Steps to build it
+
+- Create and navigate to the folder where you are going to create the empty project.
+
+- Execute `npm init`, you will be prompted to answer some information about the project (e.g. set name to _sample-vue-js_ and description to _Sample working with TypeScript and Webpack_).
+Once you have successfully fullfilled them a **package.json** file we will generated.
+
+ ```
+ npm init
+ ```
+
+- Install **webpack** as a development dependency.
+
+ ```
+ npm install webpack webpack-cli --save-dev
+ ```
+
+- Install **webpack-dev-server** locally, as a development dependency (the reason to install it locally instead of globally is for it to be easy to setup, e.g. It can be launched on a clean machine without having to install anything globally but nodejs).
+
+ ```
+ npm install webpack-dev-server --save-dev
+ ```
+
+- Let's install a list of plugins and loaders that will add powers to our webpack configuration (handling <abbr title="Cascading Style Sheets">CSS</abbr>, TypeScript...).
+
+ ```
+ npm install ts-loader fork-ts-checker-webpack-plugin css-loader file-loader html-webpack-plugin mini-css-extract-plugin url-loader  --save-dev
+ ```
+
+- Let's add two commands to our **package.json** to build and start.
+
+### ./package.json
+```diff
+  "scripts": {
++    "start": "webpack-dev-server",
++    "build": "webpack"
+  },
+
+```
+
+- Let's install locally TypeScript:
+
+ ```
+ npm install typescript@2 --save-dev
+ ```
+
+- We need as well to drop a **tsconfig.json** file in the root folder of our project
+
+### ./tsconfig.json
+ ```json
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "es6",
+    "moduleResolution": "node",
+    "declaration": false,
+    "noImplicitAny": false,
+    "sourceMap": true,
+    "noLib": false,
+    "suppressImplicitAnyIndexErrors": true
+  },
+  "compileOnSave": false,
+  "exclude": [
+    "node_modules"
+  ]
+}
+
+ ```
+
+- Since we had configured TypeScript to work with ES6, we need to install `babel` to transpile to ES5:
+
+```
+TypeScript transpile to ES6 files and Babel transpile to ES5 files
+
+      TypeScript            Babel
+.ts ============> ES6 .js =========> ES5 .js
+
+```
+
+```
+npm install babel-core babel-preset-env --save-dev
+
+```
+
+- Add `.babelrc`:
+
+### ./.babelrc
+```diff
+{
+  "presets": [
+    "env"
+  ]
+}
+
+```
```

Comment:
As last config step, I will add the `.editorcofing`. I know that it was not on previous samples, but it could be an improvement. With this file we have same config in our editor:

```
root = true

[*]
indent_size = 2
indent_style = space
insert_final_newline = true
end_of_line = crlf
charset = utf-8
trim_trailing_whitespace = true
```

Reply:
Perfect!